File: ex5/ex5_2.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    string centerSpread(string &str, int i, int j)
    {
        while (i >= 0 && j < str.size() && str[i] == str[j])
        {
            --i;
            ++j;
        }
        return str.substr(i + 1, j - i + 1 - 2);
    }
    string longestPalindrome(string s)
    {
        int n = s.size();
        int maxlen = 0;
        string res;
        for (int i = 0; i < n; ++i)
        {
            string oddstring = centerSpread(s, i, i);
            string evenstring = centerSpread(s, i, i + 1);
            string maxstring = oddstring.size() > evenstring.size() ? oddstring : evenstring;
            if (maxstring.size() > maxlen)
            {
                maxlen = maxstring.size();
                res = maxstring;
            }
        }
        return res;

    }
};

int main(void)
{

    return 0;
}

File: ex5/ex5_2.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        maxlen = 0
        res = ""

        def centerSpread(i, j):
            while i >= 0 and j < n and s[i] == s[j]:
                i -= 1
                j += 1
            return s[i+1:i+1+j-i+1-2]
        for i in range(n):
            oddstring = centerSpread(i, i)
            evenstring = centerSpread(i, i+1)
            maxstring = oddstring if len(oddstring) > len(
                evenstring) else evenstring
            if len(maxstring) > maxlen:
                maxlen = len(maxstring)
                res = maxstring
        return res